sdk/playwrighttesting/microsoft-playwright-testing/src/reporter/mptReporter.ts

```diff
@@ -113,7 +114,7 @@ class MPTReporter implements Reporter {
     if (!this.enableResultPublish) return;
     this.initializeMPTReporter();
     this.reporterUtils = new ReporterUtils(this.envVariables, config, suite);
-    if (this.isTokenValid && this.isRegionValid) {
+    if (this.isTokenValid && this.isRegionValid && this.isSameWorkspace) {
```

Comment:
NIT: we can refactor this to make single variable/state as isInitialized during init and then this variable as gating condition in all hooks.